fix: consistently set executor log options

agilgur5 opened the pull request:
<!-- markdownlint-disable MD041 -- this is rendered within existing HTML, so allow starting without an H1 -->

<!--

### Before you open your PR

- Run `make pre-commit -B` to fix codegen and lint problems (build will fail).
- [Signed-off your commits](https://github.com/apps/dco/) (otherwise the DCO check will fail).
- Used [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/).

### When you open your PR

- PR title format should also conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
- "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
- Create the PR as draft.
- Once builds are green, mark your PR "Ready for review".

When changes are requested, please address them and then dismiss the review to get it reviewed again.

-->

<!-- Does this PR fix an issue -->

Fixes #12788, Fixes https://github.com/argoproj/argo-workflows/pull/12929#discussion_r1572707894

### Motivation

<!-- TODO: Say why you made your changes. -->

- loglevel, gloglevel, and log format were inconsistently set around the codebase for executor containers
  - meaning that some had partial settings and some had no settings, depending on the command run etc

### Modifications

<!-- TODO: Say what changes you made. -->
<!-- TODO: Attach screenshots if you changed the UI. -->

- for all commands, including `emissary`, `agent`, `resource`, and `data`
  - set loglevel, gloglevel, and log format

- replace `getExecutorLogLevel` helper with `getExecutorLogOpts` helper
- add a helper in `util/cmd/glog.go` to get the gloglevel

### Verification

<!-- TODO: Say how you tested your changes. -->

Existing tests pass

<!--
### Beyond this PR

Thank you for submitting this! Have you ever thought of becoming a Reviewer or Approver on the project?

Argo Workflows is seeking more community involvement and ultimately more [Reviewers and Approvers](https://github.com/argoproj/argoproj/blob/main/community/membership.md) to help keep it viable.
See [Sustainability Effort](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) for more information.

-->

agilgur5 commented on workflow/controller/operator_test.go:
```diff
@@ -3297,11 +3297,8 @@ func TestResolveIOPathPlaceholders(t *testing.T) {
-		"--", "sh", "-c", "head -n 3 <\"/inputs/text/data\" | tee \"/outputs/text/data\" | wc -l > \"/outputs/actual-lines-count/data\"",
-	}, pods.Items[0].Spec.Containers[1].Command)
+	assert.Equal(t, append(append([]string{"/var/run/argo/argoexec", "emissary"}, woc.getExecutorLogOpts()...),
+		"--", "sh", "-c", "head -n 3 <\"/inputs/text/data\" | tee \"/outputs/text/data\" | wc -l > \"/outputs/actual-lines-count/data\""), pods.Items[0].Spec.Containers[1].Command)
```
huh, I thought `gofmt` would have something to say about this / auto-fix this 🤔

tczhao reviewed (commented):
Shall we also comment in the CLIs
that `gloglevel` is k8s go-client logging level range 0-10

agilgur5 commented:
You mean making [this line](https://github.com/argoproj/argo-workflows/blob/ed55b02992d5774e6d7ccb56bef73c27cb0c5fd2/cmd/argo/commands/root.go#L138) more specific?

I didn't change the option in this PR, just made it actually used, but could improve that for sure

agilgur5 commented:
Merged `main` for new checks per #13027

agilgur5 commented:
Rebased and fixed a good chunk of merge conflicts since this has been waiting for review 6+ months now 😕 

@tczhao @terrytangyuan could one of you review this?

> You mean making [this line](https://github.com/argoproj/argo-workflows/blob/ed55b02992d5774e6d7ccb56bef73c27cb0c5fd2/cmd/argo/commands/root.go#L138) more specific?
> 
> I didn't change the option in this PR, just made it actually used, but could improve that for sure, especially as it is [for](https://github.com/kubernetes/client-go/blob/235936510e4ea63c889836f7d6b06ddcf6dfa08e/go.mod#L31) [the `klog` fork](https://github.com/kubernetes/klog)

I can also do this in a separate PR, it's a bit orthogonal to this change
